native/physics/src/lib.rs

```diff
@@ -67,11 +69,19 @@ fn move_player(game_state: GameState, player_id: u64, x: f64, y: f64) -> GameSta
 }
 
 #[rustler::nif()]
-fn move_entities(game_state: GameState) -> GameState {
+fn move_entities(game_state: GameState, obstacles: Vec<Entity>) -> GameState {
     let mut game_state: GameState = game_state;
-
     for entity in game_state.entities.values_mut() {
+        // Avoid calculation for entities with no speed
+        if entity.speed == 0.0 {
```

Comment:
Can you add a direction condition? direction.x or direction.y should be different to 0.

Reply:
Sure I can add the check
But is not redundant? If the speed is 0, the player won't move, whether it has direction or no

Reply:
Yes, but the speed is something inherent to the entity, it is the speed that it is going to move if it is moving. Right now it is the direction that determines whether it moves or not.